src/transport_ofi.c

```diff
@@ -603,24 +610,44 @@ static inline int exchange_and_av_insert(struct fabric_info *info)
 	return ret;
     }
 
+    full_ep_len = full_ep_len + epnamelen;
+    epnamebuff = malloc(full_ep_len);
+
     alladdrs = malloc(info->npes * epnamelen);
     if(alladdrs==NULL){
 	OFI_ERRMSG("alladdrs is NULL\n");
 	return ret;
     }
 
-    ret = shmem_runtime_put("OFI", epname, epnamelen);
+#ifdef USE_ON_NODE_COMMS
+    gethostname(&epname[epnamelen], pidlen);
+#endif
+
+    ret = shmem_runtime_put("OFI", epname, full_ep_len);
     shmem_runtime_exchange();
     shmem_runtime_barrier();
 
     for(i=0; i<info->npes; i++)
     {
-	    void *tgt = alladdrs + i*epnamelen;
+	    adrlen = i*epnamelen;
+
 	    if (i == rank) {
-		    memcpy(tgt, epname, epnamelen);
+		    memcpy(epnamebuff, epname, full_ep_len);
```

Comment:
Can this overflow or walk off end end of epname?  On line 613, full_ep_len is increased beyond the size of epname, and this is memcpy, not strncpy, so it will wander past the NULL in epname.